### PR TITLE
Implement ImportStringBuilder

### PR DESCRIPTION
**Pull request by scveloso**
## Done:
- Create new ImportStringBuilder util class - when passed a ClassDecl or an ImportDecl, gives an import string body.
e.g.
```
import {VarList} from "../ast/VarList";
import {PathTable} from "./PathTable";
import {ClassDecl} from "../ast/ClassDecl";
```

- Create PathTable - a singleton that carries the paths to any type.

### CODE REVIEW

File: src/ast/ClassDecl.ts
Comment (jyoo980) [inline src/ast/ClassDecl.ts]: Just as an aside, you can get away from using string concatenation by using template literals
```TypeScript
return `${this.parentPath}/${this.className}.ts`;
```

File: src/ast/VarList.ts
Comment (jyoo980) [inline src/ast/VarList.ts:12]: Thanks for the name change; definitely makes more sense calling it`nameTypeMap`, which is exactly what it is 😄

File: src/util/PathTable.ts
Comment (jyoo980) [inline src/util/PathTable.ts]: I'm assuming you gave this an intermediate assignment of `typePath` to aid in readability?

Comment (scveloso) [reply src/util/PathTable.ts]: Yes but i dont mind changing it to just straight up return the path.

File: src/ast/ClassDecl.ts
Comment (scveloso) [reply src/ast/ClassDecl.ts]: Yeah there are some other areas where I use string interpolation instead of concatenation, i can change this for sure

File: src/util/ImportStringBuilder.ts
Comment (kiyomih) [inline src/util/ImportStringBuilder.ts:33]: For the places where we have repeated logic (e.g. getClassImportsString/ getInterfaceImportsString, is it possible to just consolidate these methods to one using a union type? e.g.:
`private static getImportsString(decl: ClassDecl | InterfaceDecl): string `

**Review by kiyomih — COMMENTED** [review]
🔥🙌🏻‼️
Looks like this was a lot of work! Ty for working on it!! 🙇🏻‍♀️
Added a comment about possibly reducing some of the duplication of logic in ImportStringBuilder

File: src/util/ImportStringBuilder.ts
Comment (scveloso) [reply src/util/ImportStringBuilder.ts:33]: I didnt know this was a thing, ill look into it!

Comment (scveloso) [reply src/util/ImportStringBuilder.ts:33]: Ok so they have to be separate because the way we get fields from one and the other is different. Class have lists of FieldDecl while Interface only has one FieldDecl. The logic is different so down the line they would also have to be separated.

**Review by jyoo980 — APPROVED** [review]
@scveloso 

LGTM, maybe hold off on merging till you get a +1 from @kiyomih

File: src/util/ImportStringBuilder.ts
Comment (kiyomih) [reply src/util/ImportStringBuilder.ts:33]: I see, oki doki- thanks for looking into it!

**Review by kiyomih — APPROVED** [review]

File: src/util/ImportStringBuilder.ts
Comment (scveloso) [reply src/util/ImportStringBuilder.ts:33]: Actually I looked at it more and there were a couple of things I could do. Check it out - I tried to reduce repeated logic as much as possible.

**Pull request merged**
